### PR TITLE
pretrained sentiment analysis added

### PR DESCRIPTION
**Pull request by Abuton**
I performed sentiment analysis on few parts of the review dataset using pre-train model specifically transformers. The small part then serves as my training dataset and i used that to build a logistic regression model to predict on all the remaining reviews. Please review and let me know what you think

### CODE REVIEW

**Review by Johnne32 — COMMENTED** [review]
In response to your previous questions, I made use of rating score as my sentiment 'key' because the reviews of the customers is directly tied to their review scores. Review scores of 1-5 were given and reviews having 4 and 5 were mostly positive reviews while 1 and 2 were mostly negative reviews. 3 showed neutral reviews which was why I excluded it from my notebook.

I have gone through this pretrained sentiment analysis notebook and I must say that I am not impressed with the outcome/method unless you are probably getting it wrong. These are my reasons;
1. In text processing, you changed 't' to 'not' (why?) and added 'can' to it. It's Portuguese we are dealing with, you know. I think the preprocessing steps need to reviewed.

2. From the first 10000 samples worked on, In the very first row of the result, the "sentiment score" was wrong. it was supposed to be positive and I don't know how others would probably be affected.

3. After performing the text classification, an accuracy of 0.8 was obtained which is low in my opinion and may contains some significant errors. When I initially used logistic regression to model this dataset, I got 0.94 as my accuracy.

This pretrained  method seem interesting but I believe it can be improved on. Well done.

**Comment by ugoiloh** [comment]
image
Well done @Abuton 

From the snap shot here, your test set is way bigger than your test set... was this an error?
As @Johnne32 pointed out, what's the justification behind replacing 't' with not, bearing in mind that you're working with untranslated Portuguese language?

Making a summary of your solution would enhance completeness and understanding of your work.
Also, tidying up for unused or error codes is necessary.

Thanks

**Comment by Abuton** [comment]
Thanks guys for your inputs. I didn't actually use the text_Preprocessing function so the not t issue was just a misunderstanding. Sorry for that I have made necessary changes to the notebook,
For the size of the data set, I just wanted to test the ability of the pretrained model and listen to your inputs I can increase it and I plan to do that next time I make a PR. 
@Johnne32 the 80% accuracy is mostly due to the small size of the training data, I will make necessary changes and make another PR. Once again Thank you guys

**Comment by mentor-mide** [comment]
Hello @Abuton, I am trying check the the code for the sentiment analysis... but it showing in like hmtl format. Please what are the model you used for analysis of the description and prediction.. I want to check to see if we can optimise the result.

**Comment by Abuton** [comment]
Hi @Ademide-Ade, I used a transformer pretrained model to generate my training data and used logistic Regression algorithm to build the sentiment model. I don't know why it not showing in the correct format though. Thanks

**Comment by mentor-mide** [comment]
Hello Sir,

Thanks... Mr John suggested I should download the code file...

That's great well-done.

So the transformer concerted the words to vectors and the logistic model
evaluated the data?

That's pretty much a tough work well done Sir.


Best regards
